libra-node/src/main_node.rs

```diff
@@ -161,19 +151,7 @@ pub fn setup_environment(node_config: &mut NodeConfig) -> LibraHandle {
                     panic!("There can be at most one validator network!");
                 }
                 // HACK: gossip relies on on-chain discovery for the eligible peers update.
-                if network_config.discovery_method != DiscoveryMethod::Onchain {
-                    let conn_mgr_reqs_tx = network_builder
-                        .conn_mgr_reqs_tx()
-                        .expect("ConnectivityManager must be installed for validator");
-                    let (simple_discovery_reconfig_subscription, simple_discovery_reconfig_rx) =
-                        gen_simple_discovery_reconfig_subscription();
-                    reconfig_subscriptions.push(simple_discovery_reconfig_subscription);
-                    let network_config_listener =
-                        ConfigurationChangeListener::new(conn_mgr_reqs_tx, role);
-                    runtime
-                        .handle()
-                        .spawn(network_config_listener.start(simple_discovery_reconfig_rx));
-                }
+                if network_config.discovery_method != DiscoveryMethod::Onchain {}
```

Comment:
Uh, is there supposed to still be something in the if?  Or are we going to just wipe both the hack and the if statement.

Reply:
Good catch. This comment needs to move into network_builder.